### DISCUSSION TITLE
nb & Helix

### DISCUSSION THREAD

**Discussion by NeotenicPrimate** [xwmx/nb#295] [Q&A]
I am trying to use nb but when I go to edit a note it defaults to opening VsCode. Is it possible to use nb with Helix? If so, does anyone have a link to how I could go about doing so?

**Comment by xwmx** [comment]
@NeotenicPrimate Helix looks really nice. Thanks for the heads up. You can use Helix as the default editor for `nb` using the following command:

```
nb set editor hx
```

Let me know if you run into any issues with it.

**Comment by PresGas** [comment]
I know this was brought up some time ago, but my question is related, @xwmx . Is there a way for nb to open up _into_ a new buffer of an existing editor like helix et al?

I guess I am envisioning this being similar to if you pick vscode/codium as your editor in that every time you do an `nb e` it would bring the note up into a new tab. However, in this case, it would be a new buffer.

**Reply by sebaschi** [reply]
tldr; no, there is no (direct) way to open _into_ a new buffer of an existing helix  editor process.

This is possible for any editor that supports opening files in existing sessions. In that case, you can write a wrapper script which uses the correct commands, and then set that script to be the editor for nb.
Helix (currently) has no mechanism to support this (opening files in a running hx instance), so it is not (directly) possible, see https://github.com/helix-editor/helix/issues/2177.
